lxd/storage_volumes_snapshot.go

```diff
@@ -0,0 +1,502 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/gorilla/mux"
+
+	"github.com/lxc/lxd/lxd/db"
+	"github.com/lxc/lxd/lxd/util"
+	"github.com/lxc/lxd/shared"
+	"github.com/lxc/lxd/shared/api"
+	"github.com/lxc/lxd/shared/version"
+)
+
+var storagePoolVolumeSnapshotsTypeCmd = Command{
+	name: "storage-pools/{pool}/volumes/{type}/{name}/snapshots",
+	post: storagePoolVolumeSnapshotsTypePost,
+	get:  storagePoolVolumeSnapshotsTypeGet,
+}
+
+var storagePoolVolumeSnapshotTypeCmd = Command{
+	name:   "storage-pools/{pool}/volumes/{type}/{name}/snapshots/{snapshotName}",
+	post:   storagePoolVolumeSnapshotTypePost,
+	get:    storagePoolVolumeSnapshotTypeGet,
+	put:    storagePoolVolumeSnapshotTypePut,
+	delete: storagePoolVolumeSnapshotTypeDelete,
+}
+
+func storagePoolVolumeSnapshotsTypePost(d *Daemon, r *http.Request) Response {
+	// Get the name of the pool.
+	poolName := mux.Vars(r)["pool"]
+
+	// Get the name of the volume type.
+	volumeTypeName := mux.Vars(r)["type"]
+
+	// Get the name of the volume.
+	volumeName := mux.Vars(r)["name"]
+
+	// Parse the request.
+	req := api.StorageVolumeSnapshotsPost{}
+	err := json.NewDecoder(r.Body).Decode(&req)
+	if err != nil {
+		return BadRequest(err)
+	}
+
+	// Get a snapshot name.
+	if req.Name == "" {
+		// i := d.cluster.ContainerNextSnapshot(volumeName)
+		i := 0
+		req.Name = fmt.Sprintf("snap%d", i)
+	}
```

Comment:
Hm, you need to implement a function that actually looks into the db and gets the highest number for the latest snapshot. This should be extremely similar to what we do for container snapshots.